src/Auth/Auth.js

```diff
@@ -101,4 +102,51 @@ export default class Auth {
     ).split(" ");
     return scopes.every(scope => grantedScopes.includes(scope));
   }
+
+  renewToken(cb) {
+    // The first parameter to checkSession allows us to specify the audience and scope.
+    // It uses the settings from when we instantiated this Auth0 WebAuth object
+    // if we omit these properties.
+    this.auth0.checkSession({}, (err, result) => {
+      if (err) {
+        console.log(`Error: ${err.error} - ${err.error_description}.`);
+      } else {
+        this.setSession(result);
+      }
+      if (cb) cb(result);
+    });
+  }
+
+  scheduleTokenRenewal() {
+    const expiresAt = JSON.parse(localStorage.getItem("expires_at"));
+    const delay = expiresAt - Date.now();
+    // Delay in milliseconds before requesting renewal.
+    // Will be 7200000 (120 minutes) immediately after login
+    // since Auth0's default token expiration is 2 hours.
+    if (delay > 0) setTimeout(() => this.renewToken(), delay);
+  }
+
+  // ----------------
+  // API Calls
+  // These calls are composed here so we can keep the access token private to this file
+  // This protects from XSS since the access token isn't accessible outside this file
+
+  getPrivate() {
+    return fetch("/private", {
```

Comment:
An attacker could just overwrite `window.fetch` and access the token that way instead?

Reply:
Perhaps. Shoot, if I was using Axios instead, I'd be referencing an import and thus overriding it wouldn't be possible. That said, I'm not sure if what I'm trying here is helpful/necessary. Unfortunately, I can find no docs on people doing this beyond @brunokrebs post. I don't see how storing in memory protects from XSS if the values in memory are just as easily accessible via JS.

Reply:
Accessing in-memory variables like `_idToken` and `_accessToken` is as easy as parsing content available at `localStorage`?

Reply:
You could do the same thing with `XMLHttpRequest` (which Axios uses behind the scenes in place of `fetch`).

I was drawn to this because I researched it recently and I feel your pain with this - there is a severe lack of best practice articles on the web on the matter, especially for single page applications. I'm by no means a security expert, but my take is that if you're using third-party scripts, including `node_modules` (even ones that are part of your build), then that's where the real problem lies, so it's just then a matter of taking steps to mitigate the risks involved.

Reply:
That's a great point @jrwebdev. I'm going to revert moving the API calls here and put them back in their original React components.